### PR TITLE
Merge multiple services with master

### PR DESCRIPTION
**Pull request by skletsun**
## Overview
Changes from `master` applied to `multiple-services` before merging back to `master`.

### Definition of Done
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/rust/ejb-app/TUTORIAL.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ This document describes how to configure and run an Exonum node with a Java serv
 ## Prerequisites
 
 Build an application following the instructions in [“How to Build”][how-to-build] section
-of the Contribution Guide.
+of the Contribution Guide. Unpack the zip archive from `exonum-java-binding/core/target` directory to some known
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/ejb-app/TUTORIAL.md]: @vitvakatu Are these instructions still correct?

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs [hunk 2]
```diff
@@ -177,17 +201,23 @@ impl JavaServiceRuntime {
...
+        mut args_builder: InitArgsBuilder,
         runtime_config: &RuntimeConfig,
+        internal_config: InternalConfig,
     ) -> InitArgsBuilder {
-        args_builder
-            .option(&format!(
+        // We do not use system library path in tests, because an absolute path to the native
+        // library will be provided at compile time using RPATH.
+        if internal_config.system_lib_path.is_some() {
+            args_builder = args_builder.option(&format!(
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: What is the point of this assignment (and mutable args_builder)? Does this assignment have any
         effect?

File: exonum-java-binding/core/rust/ejb-app/TUTORIAL.md (hunk 1 above)
Comment (vitvakatu) [reply exonum-java-binding/core/rust/ejb-app/TUTORIAL.md]: Actually, no, the correct path is `exonum-java-bingind/packaging/target`

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs (hunk 2 above)
Comment (skletsun) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: The Builder pattern in rust consumes original instance which fits great for a chain of calls but brings such tricks if we need to update a mutable argument.

File: exonum-java-binding/core/rust/ejb-app/TUTORIAL.md (hunk 1 above)
Comment (skletsun) [reply exonum-java-binding/core/rust/ejb-app/TUTORIAL.md]: Corrected the path.

File: exonum-java-binding/core/rust/ejb-app/TUTORIAL.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ This document describes how to configure and run an Exonum node with a Java serv
 ## Prerequisites
 
 Build an application following the instructions in [“How to Build”][how-to-build] section
-of the Contribution Guide. Unpack the zip archive from `exonum-java-binding/core/target` directory to some known 
+of the Contribution Guide. Unpack the zip archive from `exonum-java-bingind/packaging/target` directory to some known
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/ejb-app/TUTORIAL.md]: bingind

**Comment by dmitry-timofeev** [comment]
> Tbh, hard to say, but we'll see possible problems when merging with master :)

It is quite reasonable when the diff is reviewed from intellij, of a selection of commits (merge commit + subsequent fixups).

**Pull request merged**